airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/write/BasicFunctionalityIntegrationTest.kt

```diff
@@ -130,4 +149,179 @@ abstract class BasicFunctionalityIntegrationTest(
             },
         )
     }
+
+    /**
+     *
+     */
+    @Test
+    open fun testMidSyncCheckpointingStreamState() = runTest {
```

Comment:
We probably don't want to use `runTest` for tests that wrap the destination process? I'm assuming the delay-suppressing behavior propagates to the application context? If so there are probably a couple of processes that either loop and sleep (like TimedForcedFlush) or busy wait (like awaiting flushes at the end) that are now blocking.

Probably the latter? Because it's taking the lock and checking the checkpoint maps over and over without giving anyone else a chance to flush?

Just a guess, try it with `runBlocking`

Reply:
still gets stuck with runBlocking (and also gets stuck when I run against the docker process :/ which shouldn't be affected by any coroutines stuff)

Reply:
I'll take another look after lunch.

Reply:
pushed https://github.com/airbytehq/airbyte/pull/46749/commits/5303de5adedeb1288fd0e0c5d051ff459ac103a2 as a quickfix to get the nondocker thing working. 

actually - do you mind pairing on this tomorrow? mostly I'm curious how you go about debugging this stuff - it's not obvious to me where I should be setting breakpoints, and also I'm struggling a bit with understanding intellij's debugger interface